spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -732,4 +742,10 @@ protected Properties propertiesFromConsumerPropertyOverrides() {
 		return props;
 	}
 
+	public boolean isContainerFenced() {
```

Comment:
Again: I don't see a reason in any new API like this.
The functionality is totally connected with the `start()` and if you fail there, then something is off in your logic.
Don't give the framework users to make some choice which would be possible with this new public API.
Just throw that `IllegalStateException` from the start.
If you meet such an exception, then something has to be fixed in the target project.